File: chapter1/1-1-4.js
/*
    Define a Circle class and provide method for calculating area and perimeter
 */

//automated code testing on Alyra plateform needs french name
class Cercle {
    constructor(radius) {
        this.radius = radius;
    }
    perimetre() { return 2 * Math.PI * this.radius; }
    aire()  { return Math.PI * this.radius ** 2;}
}

//Testing class Cercle
let c = new Cercle(5);
console.log({aire: c.aire(), perimetre: c.perimetre()});

File: chapter1/1-1-4.py
import math

class Cercle:
    def __init__(self, radius):
        self.radius = radius
    def perimetre(self):
        return 2 * math.pi * self.radius
    def aire(self):
        return math.pi * pow(self.radius, 2)

# Test class Cercle
c = Cercle(5)
print({'aire': c.aire(), 'perimetre': c.perimetre()})

